src/Compilers/Core/Portable/InternalUtilities/GeneratedCodeUtilities.cs

```diff
@@ -76,6 +82,35 @@ private static bool IsGeneratedCodeFile(string filePath)
             return false;
         }
 
+        private static bool BeginsWithAutoGeneratedComment(SyntaxNode root, Func<SyntaxTrivia, bool> isComment)
```

Comment:
Sorry for the delayed comment, but seems like you can share majority of code between this method and the overload below which takes `SyntaxTree` argument.